src/Symfony/Component/Process/ExecutableFinder.php

```diff
@@ -55,6 +58,8 @@ public function find(string $name, ?string $default = null, array $extraDirs = [
         if ('\\' === \DIRECTORY_SEPARATOR) {
             $pathExt = getenv('PATHEXT');
             $suffixes = array_merge($pathExt ? explode(\PATH_SEPARATOR, $pathExt) : $this->suffixes, $suffixes);
+        } else {
+            $suffixes = array_merge($suffixes, $this->suffixes);
         }
```

Comment:
The empty suffix must always be last, and as @stof said, we need to account for the listed suffix on windows too when the `PATHEXT` variable is set. Then we have to decide on the best priority. I suggest listing the listed suffix last:

```suggestion
        if ('\\' === \DIRECTORY_SEPARATOR && $pathExt = getenv('PATHEXT')) {
            $suffixes = array_merge(explode(\PATH_SEPARATOR, $pathExt), $suffixes);
        }
        $suffixes = array_merge($suffixes, $this->suffixes);
```

Reply:
Ah, and one more thing: we should add a constructor that sets the default to the empty array when not on windows (or populates the array on windows)

Reply:
Like so, @nicolas-grekas? 1651d3f7b1590d03e0e1921369c91d3a8048cecc